File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.py
from collections import deque
def solution(k, score):
    result=[]
    tmp=deque()
    for i in range (len(score)):
        if(len(tmp)<k):
            tmp.append(score[i])
        else:
            for j in range (len(tmp)):
                if score[i]<=tmp[0]:
                    tmp.append(tmp.popleft())
                else:
                    if(min(tmp)==tmp[0]):
                        tmp.popleft()
                        tmp.append(score[i])
                        break
                    else:
                        tmp.append(tmp.popleft())
        result.append(min(tmp))
    
    return result

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    var answer = [];
    var result=[]
    for(let i=0; i<score.length; i++){
        answer.push(score[i])
        answer.sort((a,b)=>a-b)
        if(answer.length>k){
            answer.shift()
        }
        result.push(answer[0])
    }
    return result;
}